src/signature.c

```diff
@@ -409,16 +409,10 @@ GBytes *cms_sign(GBytes *content, const gchar *certfile, const gchar *keyfile, g
 
 		for (int i = 0; i < sk_X509_num(verified_chain); i++) {
 			const ASN1_TIME *expiry_time;
-			struct tm *next_month;
 			time_t now;
 			time_t comp;
 			time(&now);
-
-			next_month = gmtime(&now);
-			next_month->tm_mon += 1;
-			if (next_month->tm_mon == 12)
-				next_month->tm_mon = 0;
-			comp = timegm(next_month);
+			comp = now + 30*24*60*60;
```

Comment:
As the man page of [time()](http://man7.org/linux/man-pages/man2/time.2.html#BUGS) in its BUGS section states, one should use NULL as the function's argument and work with its return value instead. Please make a simple additional simplification by removing `now` completely and computing `comp` directly via

    time_t comp = time(NULL) + 30*24*60*60;